### PR TITLE
Merge #703 and #705 into master

### PR DESCRIPTION
**Pull request by ancorgs**
Merge #703 and ~~#704~~ #705 into master and update the dependency of libstorage-ng to reflect the point in which the needed feature was introduced in the TW branch.

Part of https://trello.com/c/EqN6jm0m/172-5-sles15-p3-1085134-cannot-install-when-using-xen-virtual-partitions-xvda1xvda2

### CODE REVIEW

**Comment by teclator** [comment]
LGTM although it merges #705 instead of #704  :)

**Comment by ancorgs** [comment]
Corrected the title and the description. I will merge as soon as Travis finishes (it's having connection problems today).

File: src/lib/y2partitioner/widgets/blk_device_edit_button.rb [hunk 1]
```diff
@@ -52,11 +52,12 @@ def actions
...
       end
 
-      # Whether the device is a partition
+      # Whether the device is a partition or an equivalent device (in terms of
+      # editing/formatting it).
       #
       # @return [Booelan]
       def partition?
-        device.is?(:partition)
+        device.is?(:partition) || device.is?(:stray_blk_device)
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/widgets/blk_device_edit_button.rb]: NP: `device.is?(:partition, :stray_blk_device)`

File: src/lib/y2partitioner/widgets/pages/stray_blk_device.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# encoding: utf-8
+
+# Copyright (c) [2017-2018] SUSE LLC
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/widgets/pages/stray_blk_device.rb]: NP: 2018

File: src/lib/y2partitioner/widgets/pages/stray_blk_device.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  module Widgets
+    module Pages
+      # A Page for a StrayBlkDevice (basically a XEN virtual partition)
+      class StrayBlkDevice < CWM::Page
+        # @return [Y2Storage::StrayBlkDevice] device the page is about
+        attr_reader :device
+
+        # Constructor
+        #
+        # @param [Y2Storage::StrayBlkDevice] device
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/widgets/pages/stray_blk_device.rb]: NP: We follow the format `@param device [Y2Storage::StrayBlkDevice]`

File: src/lib/y2storage/stray_blk_device.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
```
Comment (joseivanlopez) [inline src/lib/y2storage/stray_blk_device.rb]: NP: 2018

**Pull request merged**